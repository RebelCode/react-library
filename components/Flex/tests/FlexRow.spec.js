import React from "react";
import { render, cleanup } from "@testing-library/react";
import { FlexRow, FlexColumn } from "../index";

afterEach(cleanup);

describe("FlexRow", () => {
  const Comp = (
    <FlexRow>
      <FlexColumn>
        <a href="">One</a>
        <a href="">Two</a>
      </FlexColumn>
      <FlexColumn>
        <a href="">Three</a>
        <a href="">Four</a>
      </FlexColumn>
    </FlexRow>
  );

  it("should take a snapshot", () => {
    const { asFragment } = render(Comp);

    expect(asFragment(Comp)).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { queryByLabelText } = render(Comp);

    expect(queryByLabelText("FlexRow")).toBeTruthy();
  });
});
