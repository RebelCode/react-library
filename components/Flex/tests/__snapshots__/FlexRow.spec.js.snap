// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexRow should take a snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="FlexRow"
    style="display: flex; flex-flow: row nowrap; justify-content: flex-start; align-items: center;"
  >
    <div
      aria-label="FlexColumn"
      style="display: flex; flex-flow: column nowrap; justify-content: flex-start; align-items: center;"
    >
      <a
        href=""
      >
        One
      </a>
      <a
        href=""
      >
        Two
      </a>
    </div>
    <div
      aria-label="FlexColumn"
      style="display: flex; flex-flow: column nowrap; justify-content: flex-start; align-items: center;"
    >
      <a
        href=""
      >
        Three
      </a>
      <a
        href=""
      >
        Four
      </a>
    </div>
  </div>
</DocumentFragment>
`;
